sormas-backend/src/main/java/de/symeda/sormas/backend/externalmessage/ExternalMessageFacadeEjb.java

```diff
@@ -122,15 +131,46 @@ public class ExternalMessageFacadeEjb implements ExternalMessageFacade {
 	@EJB
 	private UserService userService;
 
+	@EJB
+	private CustomizableEnumFacadeEjb.CustomizableEnumFacadeEjbLocal customizableEnumFacade;
+
 	ExternalMessage fillOrBuildEntity(@NotNull ExternalMessageDto source, ExternalMessage target, boolean checkChangeDate) {
 
 		target = DtoHelper.fillOrBuildEntity(source, target, ExternalMessage::new, checkChangeDate);
 
 		target.setType(source.getType());
 		target.setExternalMessageDetails(source.getExternalMessageDetails());
 		target.setDisease(source.getDisease());
-		target.setDiseaseVariant(source.getDiseaseVariant());
-		target.setDiseaseVariantDetails(source.getDiseaseVariantDetails());
+		// If it is a LabMessage and it has not set a DiseaseVariant, an attempt is made to determine this from the attached TestReports.
+		if (ExternalMessageType.LAB_MESSAGE.equals(source.getType()) && source.getDiseaseVariant() == null && source.getSampleReports() != null) {
+			Set<TestReportDto> testReports = new HashSet<>();
+			source.getSampleReports().stream().map(SampleReportDto::getTestReports).forEach(testReportDtos -> testReports.addAll(testReportDtos));
+			Set<String> positiveTestedDiseaseVariants = new HashSet<>();
+			List<TestReportDto> positiveTestReports = new ArrayList<>();
+			testReports.stream()
+				.filter(
```

Comment:
should not take only the ones with disease variant **and** with details

also I think same disease variants with different details should be considered different variants. eg. you have Other with X in details is not equal to Other with Y details

Reply:
I don't get the first point, but I fixed the second one.